### ISSUE TITLE
Option to require median to return a value from passed collection

### ISSUE THREAD

**Issue by sebsebmc** [JuliaLang/Statistics.jl#98]
Python's [statistics](https://docs.python.org/3/library/statistics.html#statistics) has `median_low` and `median_high` to give the lower and higher value in the passed array if the array has an even number of elements. I do not see an analogous method in Julia or a way to specify any sort of behavior that will guarantee that the result of median comes from the underlying array. 

Of course, it is possible to check the length of the array to see if its even but that doesn't solve the problem of efficiently getting the 2 values in the array that are the median. 

It would be useful to be able to specify that I want the return value from median to be a value from the passed collection.

**Comment by mschauer** [comment]
Yes, we should expose that. A possible function name would be `medians` giving a range or a pair, or is this too subtile?

**Comment by sebsebmc** [comment]
`median_low` and `median_high` are nice in that they will always return 1 item, regardless of whether the collection has an even or odd number of elements. On the other hand returning both in one call is of course faster than having to make 2 calls.

**Comment by nalimilan** [comment]
See also previous discussion at https://github.com/JuliaLang/julia/pull/19359. I'd rather add arguments to `median` (like `tie=first` or `middle=first`) rather than new functions. The new PR is https://github.com/JuliaLang/julia/pull/30329, but I had some remaining objections at https://github.com/JuliaLang/julia/pull/30329#issuecomment-456795549, which will probably require some design discussion again. Help welcome to revive it!
